Add file create/read/delete code

igorveselinovic opened the pull request:
Resolves #17 

Adds one new handler for the reading, and uses existing handlers for create and delete. Also added/updated relevant unit tests.

There were also a couple of small bugs/naming issues to fix.

igorveselinovic committed:
Add creation and deletion of content file

* Create an HTML file for a conversation in the POST handler and delete
  the file in the DELETE handler.
* Fix a bug in the POST conversation handler that wasn't checking that
  the request body had the "avatar_url" field.
* Fix a log message to have all the fields for a conversation PATCH.

igorveselinovic committed:
Add variable for holding content directory

igorveselinovic committed:
Add GET handler for conversation content

Reads the file that should've been created by a POST conversation
request and writes it to the response as text/html.

igorveselinovic committed:
Update existing unit tests to test for file

The POST conversation test checks that the file was created and the
DELETE conversation test checks that the file was deleted.

igorveselinovic committed:
Attach GET content handler to server

igorveselinovic committed:
Add check for session user pending status

Pending conversation member should not be able to access conversation
content.

igorveselinovic committed:
Add unit tests for GET content handler

igorveselinovic committed:
Fix unit test names to be consistent

igorveselinovic committed:
Update README with content additions

Thao-Tran commented on handlers/content_handlers.go:
```diff
@@ -0,0 +1,65 @@
+		return
+	}
+
+	data, err := ioutil.ReadFile(filePath)
```
I think the file handling stuff should be in methods under a `Directory` (or something) struct object that just has a field for the directory and the `Env` struct should have a field for that `Directory` struct, so we keep it consistent with the database and rest client stuff.

igorveselinovic replied:
Sure, I'll do that.

Thao-Tran commented on handlers/conversation_handlers.go:
```diff
@@ -40,7 +42,13 @@ func (env *Env) PostConversationHandler(w http.ResponseWriter, r *http.Request)
+		internalServerError(w, err)
+		return
+	}
+	file.Close()
```
Same thing for the file stuff.

Thao-Tran commented on handlers/conversation_handlers.go:
```diff
@@ -151,6 +156,12 @@ func (env *Env) DeleteConversationHandler(w http.ResponseWriter, r *http.Request
+		internalServerError(w, err)
+		return
+	}
+
```
File stuff.

Thao-Tran commented on handlers/content_handlers_test.go:
```diff
@@ -0,0 +1,118 @@
+	var conversationID int64 = 1
+	filePath := path.Join(contentDir, fmt.Sprintf("%d.html", conversationID))
+	f, _ := os.Create(filePath)
+	f.WriteString("hello world")
```
Since the content is an attribute of the test, I think it should be set using that.

Thao-Tran commented on handlers/conversation_handlers.go:
```diff
@@ -27,7 +29,7 @@ func (env *Env) PostConversationHandler(w http.ResponseWriter, r *http.Request)
 	}
 
-	if reqConversation.Name == "" || reqConversation.Description == nil {
+	if reqConversation.Name == "" || reqConversation.Description == nil || reqConversation.AvatarURL == nil {
```
Like we discussed, `Description` and `AvatarURL` should be optional, so the logic needs to be updated to set them to an empty string if they weren't provided.

Thao-Tran commented on handlers/utils.go:
```diff
@@ -8,6 +8,11 @@ import (
+)
+
+var (
+	contentDir = os.Getenv("ETHER_CONTENT_DIR")
```
File stuff.

igorveselinovic committed:
Make some POST conversation fields optional

"description" and "avatar_url" are both optional fields in the request
body. Add a unit test for omitting these optional fields.

igorveselinovic committed:
Encapsulate filesystem logic in a struct

Made methods for create, read (full file), write (full file), and
delete.

igorveselinovic committed:
Fix and update README
